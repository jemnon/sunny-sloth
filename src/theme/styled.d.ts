import 'styled-components';
import {
  Breakpoints,
  Colors,
  ColorMode,
  FocusColors,
  Fonts,
  FontSizes,
  Header,
  HoverColors,
  MaxWidths,
  Spacing,
  ZIndex,
} from '../types/theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    breakpoints: Breakpoints;
    colors: Colors;
    focusColors: FocusColors;
    fonts: Fonts;
    fontSizes: FontSizes;
    header: Header;
    hoverColors: HoverColors;
    maxWidths: MaxWidths;
    spacing: Spacing;
    transition: string;
    zIndex: ZIndex;
  }
}
