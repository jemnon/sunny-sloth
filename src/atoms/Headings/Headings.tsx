import styled, { css } from 'styled-components';
import { up } from 'styled-breakpoints';

const sharedCSS = css`
  font-weight: 900;
  font-family: ${({ theme }): string => theme.fonts.noto};

  color: ${({ theme }): string =>
    theme.colorMode === 'dark' ? theme.colors.white : theme.colors.nearBlack};

  margin-bottom: 0.35em;
`;

export const H1 = styled.h1`
  ${sharedCSS};

  font-size: ${({ theme }): string => theme.fontSizes.f10};

  ${up('sm')} {
    font-size: ${({ theme }): string => theme.fontSizes.f11};
  }

  ${up('lg')} {
    font-size: ${({ theme }): string => theme.fontSizes.f13};
  }
`;

export const H2 = styled.h2`
  ${sharedCSS};

  font-size: ${({ theme }): string => theme.fontSizes.f9};

  ${up('sm')} {
    font-size: ${({ theme }): string => theme.fontSizes.f10};
  }

  ${up('lg')} {
    font-size: ${({ theme }): string => theme.fontSizes.f12};
  }
`;

export const H3 = styled.h3`
  ${sharedCSS};

  font-size: ${({ theme }): string => theme.fontSizes.f8};

  ${up('sm')} {
    font-size: ${({ theme }): string => theme.fontSizes.f9};
  }

  ${up('lg')} {
    font-size: ${({ theme }): string => theme.fontSizes.f11};
  }
`;

export const H4 = styled.h4`
  ${sharedCSS};

  font-size: ${({ theme }): string => theme.fontSizes.f7};

  ${up('sm')} {
    font-size: ${({ theme }): string => theme.fontSizes.f8};
  }

  ${up('lg')} {
    font-size: ${({ theme }): string => theme.fontSizes.f10};
  }
`;

export const H5 = styled.h5`
  ${sharedCSS};

  font-size: ${({ theme }): string => theme.fontSizes.f6};

  ${up('sm')} {
    font-size: ${({ theme }): string => theme.fontSizes.f7};
  }

  ${up('lg')} {
    font-size: ${({ theme }): string => theme.fontSizes.f9};
  }
`;

export const H6 = styled.h6`
  ${sharedCSS};

  font-size: ${({ theme }): string => theme.fontSizes.f5};

  ${up('sm')} {
    font-size: ${({ theme }): string => theme.fontSizes.f6};
  }

  ${up('lg')} {
    font-size: ${({ theme }): string => theme.fontSizes.f8};
  }
`;
