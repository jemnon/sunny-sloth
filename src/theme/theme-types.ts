export interface Breakpoints {
  sm: string;
  md: string;
  lg: string;
  xlg: string;
}

export interface Fonts {
  lato: string;
  latoBold: string;
  latoItalic: string;
  noto: string;
  notoBold: string;
}

export interface Colors {
  cream: string;
  darkTeal: string;
  teal: string;
  lightBrown: string;
  nearBlack: string;
  nearWhite: string;
  darkOrange: string;
  orange: string;
  white: string;
  red: string;
  green: string;
}

export type ColorMode = 'dark' | 'light';

export interface MaxWidths {
  sm: string;
  md: string;
  lg: string;
}

export interface Spacing {
  sm1: string;
  sm2: string;
  sm3: string;
  sm4: string;
  md1: string;
  md2: string;
  md3: string;
  md4: string;
  lg1: string;
  lg2: string;
  lg3: string;
  lg4: string;
}

export interface ZIndex {
  z0: string;
  z1: string;
  z2: string;
  z3: string;
  z4: string;
  z5: string;
  z6: string;
  z999: string;
  z9999: string;
}
