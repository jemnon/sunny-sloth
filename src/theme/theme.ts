const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
