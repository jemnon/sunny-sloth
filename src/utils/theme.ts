import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
