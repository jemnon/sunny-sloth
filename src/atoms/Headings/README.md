# Headings

### Description
Component description

```js
import Headings from '@fenderdigital/ui/FILL IN TYPEHeadings';
```

### Props
* `propName` - property name and description 
