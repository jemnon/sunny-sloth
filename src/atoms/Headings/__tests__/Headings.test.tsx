import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { H1, H2, H3, H4, H5, H6 } from '../Headings';
import theme from '../../../theme';

describe('<Headings />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <H1 />
      <H2 />
      <H3 />
      <H4 />
      <H5 />
      <H6 />
    </ThemeProvider>,
  );
  it('exists', (): void => {
    expect(H1).toBeDefined();
    expect(H2).toBeDefined();
    expect(H3).toBeDefined();
    expect(H4).toBeDefined();
    expect(H5).toBeDefined();
    expect(H6).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
