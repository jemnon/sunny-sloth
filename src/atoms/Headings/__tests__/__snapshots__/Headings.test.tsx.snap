// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headings /> renders defaults correctly 1`] = `
Array [
  <h1
    className="sc-bdfBwQ kgBeFo"
  />,
  <h2
    className="sc-gsTCUz jHSrHZ"
  />,
  <h3
    className="sc-dlfnbm hYvFYw"
  />,
  <h4
    className="sc-hKgILt jYwWFD"
  />,
  <h5
    className="sc-eCssSg bpmnQu"
  />,
  <h6
    className="sc-jSgupP gOyKjb"
  />,
]
`;
